src/components/modules/weeklyBanner/WeeklyBanner.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from 'react';
+import { ThemeProvider } from 'emotion-theming';
+import { Button, LinkButton, buttonReaderRevenue } from '@guardian/src-button';
+import { brand } from '@guardian/src-foundations/themes';
+import { SvgGuardianLogo } from '@guardian/src-brand';
+import { SvgClose } from '@guardian/src-icons';
+import {
+    banner,
+    contentContainer,
+    topLeftComponent,
+    heading,
+    paragraph,
+    buttonTextDesktop,
+    buttonTextMobileTablet,
+    siteMessage,
+    bottomRightComponent,
+    packShot,
+    iconPanel,
+    closeButton,
+    logoContainer,
+} from './weeklyBannerStyles';
+
+type WeeklyBannerProps = {
+    subscriptionUrl: string;
+    signInUrl: string;
+};
+
+export const WeeklyBanner: React.FC<WeeklyBannerProps> = ({
+    subscriptionUrl,
+    signInUrl,
+}: WeeklyBannerProps) => {
+    const [showBanner, closeBanner] = useState(true);
+    return (
+        <>
+            {showBanner ? (
+                <section
+                    id="js-site-message--subscription-banner"
```

Comment:
Are these IDs needed still? The `js-` prefix made them jump out at me (maybe they're left over from the un-Reacty frontend code?)

Reply:
Thanks for pointing that out - I will remove them.